File: app/src/main/java/com/hippo/ehviewer/gallery/EhGalleryProvider.java
/*
 * STUB (LANraragi: E-Hentai gallery provider removed)
 * LANraragi uses LRRGalleryProvider instead.
 */
package com.hippo.ehviewer.gallery;

import android.content.Context;
import com.hippo.ehviewer.client.data.GalleryInfo;

/**
 * EhGalleryProvider — STUB.
 * E-Hentai SpiderQueen-based gallery provider has been removed.
 */
public class EhGalleryProvider extends GalleryProvider2 {

    public EhGalleryProvider(Context context, GalleryInfo galleryInfo) {
        // No-op: E-Hentai gallery provider removed
    }

    @Override
    public int size() { return 0; }

    @Override
    protected void onRequest(int index) {}

    @Override
    protected void onForceRequest(int index) {}

    @Override
    protected void onCancelRequest(int index) {}

    @Override
    public String getError() { return "E-Hentai provider removed"; }

    @Override
    public String getImageFilename(int index) { return ""; }

    @Override
    public boolean save(int index, com.hippo.unifile.UniFile file) { return false; }

    @Override
    public com.hippo.unifile.UniFile save(int index, com.hippo.unifile.UniFile dir, String filename) { return null; }
}

File: app/src/main/java/com/hippo/ehviewer/gallery/EhGalleryProvider.kt
/*
 * STUB (LANraragi: E-Hentai gallery provider removed)
 * LANraragi uses LRRGalleryProvider instead.
 */
package com.hippo.ehviewer.gallery

import android.content.Context
import com.hippo.ehviewer.client.data.GalleryInfo
import com.hippo.unifile.UniFile

/**
 * EhGalleryProvider — STUB.
 * E-Hentai SpiderQueen-based gallery provider has been removed.
 */
class EhGalleryProvider(context: Context, galleryInfo: GalleryInfo) : GalleryProvider2() {

    override fun size(): Int = 0

    override fun onRequest(index: Int) {}

    override fun onForceRequest(index: Int) {}

    override fun onCancelRequest(index: Int) {}

    override fun getError(): String = "E-Hentai provider removed"

    override fun getImageFilename(index: Int): String = ""

    override fun save(index: Int, file: UniFile): Boolean = false

    override fun save(index: Int, dir: UniFile, filename: String): UniFile? = null
}
